module/item/item.js

```diff
@@ -0,0 +1,626 @@
+                    weaponData.canAddTension = true;
+                    weaponData.tensionDefault = "yes";
+                } else {
+                    dmgRoll += " + 1" + td;
+                }
+                dmgSnd = data.burstAttack.dmgSnd;
+                flavorAdd = " on burst fire";
+                break;
+            case "fullAuto":
+                dmgSnd = data.autoAttack.dmgSnd;
+                dmgRoll += " + " + fullAutoDamage.toFixed(0) + td;
+                flavorAdd = " on full auto";
+                break;
+        }
+
+        const r = new CONFIG.Dice.DamageRoll(dmgRoll, this.actor.data.data);
+
+        const configured = await r.configureDialog({
+            title: `Damage from ${this.data.name}`,
+            defaultRollMode: game.settings.get("core", "rollMode"),
+            weaponData
+        });
+        if (configured === null) {
+            return;
+        }
+
+        let messageData = {
+            speaker: ChatMessage.getSpeaker({ actor: this.actor }),
+            flavor: "Done damage using " + this.data.name + flavorAdd
+        };
+
+        if (dmgSnd) { // Check if the sound is set, then check if it exists
+            const resp = await fetch(dmgSnd, { method: 'HEAD' });
+            if (resp.ok) {
+                messageData.sound = dmgSnd;
+            } else {
+                ui.notifications.warn("Damage sound path for " + this.data.name + " could not be resolved: " + dmgSnd);
+            }
+        }
+
+        return r.toMessage(messageData);
+    }
+
+    async consume() {
+        const remainingCount = parseInt(this.data.data.quantity);
+        if (remainingCount < 1) {
+            return ui.notifications.warn("You dont have any more of these items to consume!");
+        }
+
+        await this.update({
+            "data.quantity": remainingCount - 1
+        });
+        return ui.notifications.info(`Item '${this.data.name}' was consumed, ${remainingCount - 1} usages remain.`);
+    }
+
+    /**
+     * Find the ammo item for this weapon, if any exists
+     * @returns {ItemSg}
+     */
+    findAmmunition() {
+        if (!this.consumesAmmunition) {
+            return null;
+        }
+        return this.actor.items.get(this.data.data.ammo.target);
+    }
+
+    async reloadWeapon() {
+        const item = this;
+        const data = item.data.data;
+
+        if (data.ammo.value === data.ammo.max) {
+            return ui.notifications.info("Weapon is already reloaded.");
+        }
+
+        const ammoItem = item.findAmmunition();
+        if (!ammoItem) {
+            // If no ammo item is found, check if the target matches the standard action reload value or is empty for a reload that doesn't consume separate items
+            if (data.ammo.target === CONFIG.SGRPG.actionReloadValue || data.ammo.target === null) {
+                if (data.ammo.extraMags === null || data.ammo.extraMags === undefined || data.ammo.extraMags <= -2) {
```

Comment:
Not sure why -2 seems like magic constant. If it has some meaning it should be named, reusable variable